tests/test_analysis.py

```diff
@@ -803,6 +803,39 @@ def test_multitaper(self):
         self.assertEqual(sgram.shape[0], self.win_t*self.fs // 2 + 1) # correct freq. bin count
         self.assertEqual(sgram.shape[-1], self.x2.shape[-1]) # correct channel output count
 
+    def test_tfspec(self):
+        num_points = 5000
+        T = 5
+        t = np.linspace(0,T,num_points)
+        test_data = np.zeros((t.shape[0],2))
+        test_data[:,0] = 1.5*np.sin(2*np.pi*10*t)
+        test_data[:,1] = 2*np.cos(2*np.pi*30*t)
+        test_data[t>=T/2,0] = 2*np.sin(2*np.pi*5*t[t<=T/2])
+        test_data[t>=T/2,1] = 1*np.cos(2*np.pi*15*t[t<=T/2])
+        
+        NW = 1
+        BW = 1
+        fs = num_points/T
+        dn = 0.1
+        fk = 50
+        n, p, k = aopy.precondition.convert_taper_parameters(NW, BW)
+        f_spec,t_spec,spec = aopy.analysis.tfspec(test_data, n, p, k, fs, dn, fk, pad=2, ref=False)
+        
+        fig,ax=plt.subplots(1,4,figsize=(8,2),tight_layout=True)
+        ax[0].plot(t,test_data[:,0],linewidth=0.2)
+        ax[0].plot(t,test_data[:,1],linewidth=0.2)
+        ax[0].set(xlabel='Time (s)',ylabel='Amplitude',title='Signals')
+        ax[1].imshow((spec[:,:,0]),aspect='auto',origin='lower',extent=[0,T,0,f_spec[-1]])
+        ax[1].set(ylabel='Frequency (Hz)',xlabel='Time [s]',title='Spectrogram (ch1)')
+        ax[2].imshow((spec[:,:,1]),aspect='auto',origin='lower',extent=[0,T,0,f_spec[-1]])
+        ax[2].set(ylabel='Frequency (Hz)',xlabel='Time [s]',title='Spectrogram (ch2)')
+        ax[3].plot(f_spec,spec[:,10,0],'-',label='ch 1')
+        ax[3].plot(f_spec,spec[:,10,1],'-',label='ch 2')
+        ax[3].set(ylabel='Power',xlabel='Frequency (Hz)',xlim=(0,50),title='Power spectral')
+        ax[3].legend(frameon=False, fontsize=7)
+        filename = 'tfspec.png'
```

Comment:
for plotting you can use https://analyze.readthedocs.io/en/latest/visualization.html?highlight=tfr#aopy.visualization.base.plot_tfr